src/app/components/shell/router-helpers/page-routes.tsx

```diff
@@ -0,0 +1,114 @@
+import React from 'react';
+import {useParams, Routes, Route, Navigate} from 'react-router-dom';
+import usePageData from '~/helpers/use-page-data';
+import FlexPage, {
+    FlexPageData,
+    LayoutUsingData
+} from '~/pages/flex-page/flex-page';
+import useLayoutContext from '~/contexts/layout';
+import {assertDefined} from '~/helpers/data';
+import {GeneralPageFromSlug} from '~/pages/general/general';
+import {ImportedPage} from './page-loaders';
+import {RouteAsPortalOrNot} from './portal-page-routes';
+
+type PageData = FlexPageData & {
+    meta: {slug: string};
+};
+
+export function HomePage() {
+    const data = usePageData<PageData>('pages/home', true);
+
+    if (!data) {
+        return null;
+    }
+    return <FlexPageUsingItsOwnLayout data={data} />;
+}
+
+export function ErrataRoutes() {
+    return (
+        <Routes>
+            <Route index element={<ImportedPage name="errata-summary" />} />
+            <Route
+                path="/form/"
+                element={<ImportedPage name="errata-form" />}
+            />
+            <Route path="/*" element={<ImportedPage name="errata-detail" />} />
+        </Routes>
+    );
+}
+
+export function DetailsRoutes() {
+    return (
+        <Routes>
+            <Route index element={<Navigate to="/subjects" replace />} />
+            <Route
+                path="/books/:title"
+                element={<ImportedPage name="details" />}
+            />
+            <Route
+                path="/:title"
+                element={<RedirectToCanonicalDetailsPage />}
+            />
+        </Routes>
+    );
+}
+
+export function OtherPageRoutes() {
+    const dir = assertDefined(useParams().dir);
+
+    if (['books', 'textbooks'].includes(dir)) {
+        return (
+            <Routes>
+                <Route
+                    path="/:title"
+                    element={<RedirectToCanonicalDetailsPage />}
+                />
+            </Routes>
+        );
+    }
+
+    if (dir === 'home') {
+        return <Navigate to="/" replace />;
+    }
+
+    return <RouteAsPortalOrNot />;
+}
+
+
+// There are a couple of pages whose names in the CMS don't match their osweb urls
+const mismatch: Record<string, string> = {
+    press: 'news',
+    'edtech-partner-program': 'openstax-ally-technology-partner-program'
```

Comment:
Is this working for you? On my end I'm seeing it 404 this page, when before it loaded.

Reply:
Yeah if I remove the /general dir then they load

Reply:
Ahh, I wasn't using the "/general". I think that should be considered obsolete. General pages can be loaded without it. If we do want to have that, I'll need to put a route back in for it; the issue you're seeing is that it's being treated as a portal (the flex page for which doesn't exist).

Reply:
I can put a redirect in (/general/something -> /something). That's very much the preferred address.